core/src/storage/device.py

```diff
@@ -344,3 +349,29 @@ def get_hide_passphrase_from_host() -> bool:
     Whether we should hide the passphrase from the host.
     """
     return common.get_bool(_NAMESPACE, _HIDE_PASSPHRASE_FROM_HOST)
+
+
+if utils.USE_THP:
+
+    def get_device_secret() -> bytes:
+        """
+        Device secret is used to derive keys that are independent of the seed.
+        """
+        device_secret = common.get(_NAMESPACE, _DEVICE_SECRET)
+        if not device_secret:
+            from trezor.crypto import random
+
+            device_secret = random.bytes(16, True)
+            common.set(_NAMESPACE, _DEVICE_SECRET, device_secret)
+        return device_secret
+
+    def get_cred_auth_key_counter() -> bytes:
+        counter = common.get(_NAMESPACE, _CRED_AUTH_KEY_COUNTER)
+        if counter is None:
+            counter = (0).to_bytes(4, "big")
+            common.set(_NAMESPACE, _CRED_AUTH_KEY_COUNTER, counter)
```

Comment:
No need to set the 0 value in storage. Most people will probably never invalidate their pairing credentials on Trezor, so no need to waste space. The code can be simplified:
```py
return common.get(_NAMESPACE, _CRED_AUTH_KEY_COUNTER) or bytes(4)
```

Reply:
ACK. Fixed in [5dce7cc](https://github.com/trezor/trezor-firmware/pull/3861/commits/5dce7cc836524bdc93f590bb8915c7098e3d1d76)